Applications/Basic/CaptureHDRLoop/CaptureHDRLoop.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+This example shows how to acquire HDR images from the Zivid camera in a loop
+(while actively changing some HDR settings).
+*/
+
+#include <Zivid/Zivid.h>
+
+#include <iostream>
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to the camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Configuring the camera settings" << std::endl;
+        std::vector<std::vector<Zivid::Settings>> settingsHDRVector;
+
+        Zivid::Settings settings;
+        settings.set(Zivid::Settings::Brightness{ 1 })
+            .set(Zivid::Settings::Gain{ 1 })
+            .set(Zivid::Settings::Bidirectional{ false })
+            .set(Zivid::Settings::Filters::Contrast::Enabled::yes)
+            .set(Zivid::Settings::Filters::Contrast::Threshold{ 5 })
+            .set(Zivid::Settings::Filters::Gaussian::Enabled::yes)
+            .set(Zivid::Settings::Filters::Gaussian::Sigma{ 1.5 })
+            .set(Zivid::Settings::Filters::Outlier::Enabled::yes)
+            .set(Zivid::Settings::Filters::Outlier::Threshold{ 5 })
+            .set(Zivid::Settings::Filters::Reflection::Enabled::yes)
+            .set(Zivid::Settings::Filters::Saturated::Enabled::yes)
+            .set(Zivid::Settings::BlueBalance{ 1.081 })
+            .set(Zivid::Settings::RedBalance{ 1.709 });
+
+        std::vector<size_t> iris{ 10U, 20U, 30U };
+        std::vector<std::vector<int>> exposureTimeFrames{ { 10000, 90000, 40000 },
+                                                          { 40000, 10000, 90000 },
+                                                          { 90000, 40000, 10000 } };
+
+        for(size_t i = 0; i < 3; ++i)
+        {
+            std::vector<Zivid::Settings> settingsHDR;
+            for(size_t j = 0; j < 3; ++j)
+            {
+                settingsHDR.push_back(
+                    settings.set(Zivid::Settings::Iris{ iris[j] })
+                        .set(Zivid::Settings::ExposureTime{ std::chrono::microseconds{ exposureTimeFrames[i][j] } }));
+            }
+            settingsHDRVector.push_back(settingsHDR);
```

Comment:
Here push_back should be ok, although I don't see why a vector is needed here. We're not referencing this vector of frame settings outside of the loop.

Reply:
Is this what you would have done?
`const` auto hdrFrame = Zivid::HDR::capture(camera, `settingsHDR);`
I thought it was nice for educational purposes to show how to create a vector of multiple HDR settings. Perhaps it is not needed as you say.
To be honest, I forgot what we agreed the purpose of this sample is to be. I remember that we've decided to keep CaptureHDRLoop and CaptureHDRCompleteSettings but I don't remember what we've agreed to emphesize in each of them.

Reply:
I don't recall exactly either. However, I believe you argued that you want an example which shows how you can hard code all settings. I argued we needed an example which will read from .yml. Maybe the following can be a compromise:

CaptureHDRCompleteSettings.cpp will set settings in code, for one HDR (multiple frames).
CaptureHDRLoop.cpp will read a set of settings from .yml for each HDR in the loop. I can update the sample as I believe it should be.

I believe this could be clearer, and cover both of our wishes.